exonum/src/storage/proof_map_index/tests.rs

```diff
@@ -384,7 +384,7 @@ fn check_map_multiproof<K, V>(
     let proof = proof.check().unwrap();
     assert_eq!(proof.merkle_root(), table.merkle_root());
     assert_eq!(missing_keys.iter().collect::<Vec<&_>>(), {
-        let mut actual_keys = proof.missing_keys();
+        let mut actual_keys = proof.missing_keys().collect::<Vec<_>>();
```

Comment:
Partially off-topic: The code of `check_map_proof` is overly complicated, it could be simplified as

```rust
fn check_map_multiproof<K, V>(
    proof: MapProof<K, V>,
    keys: Vec<K>,
    table: &ProofMapIndex<&mut Fork, K, V>,
) where
    K: ProofMapKey + Eq + Ord + Debug,
    V: StorageValue + Eq + Ord + Debug,
{
    let mut entries: Vec<(K, V)> = Vec::new();
    let mut missing_keys: Vec<K> = Vec::new();

    for key in keys {
        if table.contains(&key) {
            let value = table.get(&key).unwrap();
            entries.push((key, value));
        } else {
            missing_keys.push(key);
        }
    }

    let proof = proof.check().unwrap();
    assert_eq!(proof.merkle_root(), table.merkle_root());
    assert_eq!(
        missing_keys.iter().collect::<BTreeSet<_>>(),
        proof.missing_keys().collect::<BTreeSet<_>>()
    );
    assert_eq!(
        entries
            .iter()
            .map(|&(ref k, ref v)| (k, v))
            .collect::<BTreeSet<_>>(),
        proof.entries().collect::<BTreeSet<_>>()
    );
}
```

Reply:
Not going to fix that in this PR. I've created ECR-2294.